src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -306,41 +310,36 @@ private Document CreateNewDocument(Document doc, string key, object fieldVal)
                 case BlittableJsonReaderObject nested:
                     return new Document
                     {
-                        Id = doc.Id,
+                        Id = _context.GetLazyString(doc.Id),
                         ChangeVector = doc.ChangeVector,
                         Data = nested,
```

Comment:
`nested` comes from the original document which you're about to dispose so it will invalidate it as well